hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/NettyRpcServer.java

```diff
@@ -214,4 +228,21 @@ public int getNumOpenConnections() {
     // allChannels also contains the server channel, so exclude that from the count.
     return channelsCount > 0 ? channelsCount - 1 : channelsCount;
   }
+
+  private synchronized void initSSL(ChannelPipeline p, boolean supportPlaintext)
```

Comment:
Oh, this one is shared in NettyRpcServer, not per connection, so synchronized is needed if we share some common fields which are not thread safe(but seems not...)

I think we should share the SslContext stuff? That means, when constructing the NettyRpcServer, we check whether TLS is enabled, if so, we create the SslContext instance. So then when initializing a Channel, we just need to call the createSSLEngine method, which should be thread safe so we do not need synchronized here.